src/verilog/verilog_interfaces.cpp

```diff
@@ -99,11 +99,11 @@ void verilog_typecheckt::check_module_ports(
 
     ports[nr].set("#name", base_name);
     ports[nr].id(ID_symbol);
-    ports[nr].set(ID_identifier, identifier);
+    ports[nr].identifier(identifier);
     ports[nr].set(ID_C_source_location, declarator.source_location());
-    ports[nr].set(ID_type, port_symbol->type);
-    ports[nr].set(ID_input, port_symbol->is_input);
-    ports[nr].set(ID_output, port_symbol->is_output);
+    ports[nr].type() = port_symbol->type;
+    ports[nr].input(port_symbol->is_input);
+    ports[nr].output(port_symbol->is_output);
```

Comment:
It would be nice to have a constructor to make sure necessary members are not forgotten upon initialisation.